azkaban-web-server/src/main/java/cloudflow/services/ExecutionParameters.java

```diff
@@ -0,0 +1,155 @@
+  private Boolean notifyOnFirstFailure;
+  private Boolean notifyFailureOnExecutionComplete;
+  private ConcurrentOption concurrentOption;
+  private Map<String, Map<String, Object>> properties;
+
+  public ExecutionParameters() {
+    this.failureAction = FailureAction.FINISH_CURRENTLY_RUNNING;
+    this.concurrentOption = ConcurrentOption.CONCURRENT_OPTION_SKIP;
+    this.notifyFailureOnExecutionComplete = false;
+    this.notifyOnFirstFailure = true;
+  }
+
+  public String getFlowId() {
+    return flowId;
+  }
+
+  public void setFlowId(String flowId) {
+    this.flowId = flowId;
+  }
+
+  public Integer getFlowVersion() {
+    return flowVersion;
+  }
+
+  public void setFlowVersion(Integer flowVersion) {
+    this.flowVersion = flowVersion;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  public String getExperimentId() {
+    return experimentId;
+  }
+
+  public void setExperimentId(String experimentId) {
+    this.experimentId = experimentId;
+  }
+
+  public String getSubmitUser() {
+    return submitUser;
+  }
+
+  public void setSubmitUser(String submitUser) {
+    this.submitUser = submitUser;
+  }
+
+  public FailureAction getFailureAction() {
+    return failureAction;
+  }
+
+  public void setFailureAction(FailureAction failureAction) {
+    this.failureAction = failureAction;
+  }
+
+  public Boolean isNotifyOnFirstFailure() {
+    return notifyOnFirstFailure;
+  }
+
+  public void setNotifyOnFirstFailure(Boolean notifyOnFirstFailure) {
+    this.notifyOnFirstFailure = notifyOnFirstFailure;
+  }
+
+  public Boolean isNotifyFailureOnExecutionComplete() {
+    return notifyFailureOnExecutionComplete;
+  }
+
+  public void setNotifyFailureOnExecutionComplete(Boolean notifyFailureOnExecutionComplete) {
+    this.notifyFailureOnExecutionComplete = notifyFailureOnExecutionComplete;
+  }
+
+  public ConcurrentOption getConcurrentOption() {
+    return concurrentOption;
+  }
+
+  public void setConcurrentOption(ConcurrentOption concurrentOption) {
+    this.concurrentOption = concurrentOption;
+  }
+
+  public Map<String, Map<String, Object>> getProperties() {
+    return properties;
+  }
+
+  public void setProperties(Map<String, Map<String, Object>> properties) {
+    this.properties = properties;
+  }
+
+  public enum FailureAction {
+    FINISH_CURRENTLY_RUNNING("finishCurrent"),
+    CANCEL_ALL("cancelImmediately"),
+    FINISH_ALL_POSSIBLE("finishPossible");
+
+    private final String name;
+    private FailureAction(String name) {
+            this.name = name;
+        }
+
+    public String getName() {
+            return this.name;
+        }
+
+    @JsonCreator
+    public static FailureAction valueFromName(String name) {
+      for(FailureAction action: FailureAction.values()) {
+        if(action.name.equals(name)) {
+          return action;
+        }
+      }
```

Comment:
We have the choice of requesting as value (literally) of failureAction param either 
1- FINISH_CURRENTLY_RUNNING which can be processed with valueOf()
2- finishCurrent which has to be processed with valueFromName()
I chose option 2 because that's what's used today and its easier to type. I'm sticking with current Azkaban values because this might change completely in CF but a 3rd option is to define better names now.

Reply:
Talked offline. We agreed on using the strings associated with the enum values.